Tests/WalletCoreTests/WalletConfigurationTests/ConfigurationLoaderTests.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  ConfigurationLoaderTests.swift
+//  
+//
+//  Created by Grigory on 20.6.23..
+//
+
+import XCTest
+@testable import WalletCore
+
+final class ConfigurationLoaderTests: XCTestCase {
+    
+    let mockAPI = MockAPI<RemoteConfiguration>()
+    
+    override func setUp() {
+        super.setUp()
+        mockAPI.reset()
+    }
+    
+    func testConfigurationLoaderStartLoading() async throws {
+        // GIVEN
+        mockAPI.entity = .configuration()
+        let loader = ConfigurationLoader(api: mockAPI)
+        
+        // WHEN
+        _ = try await loader.fetch()
+    
+        // THEN
+        XCTAssertEqual(mockAPI.sendMethodCalledTimes, 1)
+    }
+    
+    func testConfigurationLoaderDoesntStartLoadingIfAlreadyLoading() async throws {
+        // GIVEN
+        mockAPI.entity = .configuration()
+        let loader = ConfigurationLoader(api: mockAPI)
+
+        // WHEN
+        async let task = loader.fetch()
+        async let task2 = loader.fetch()
+        async let task3 = loader.fetch()
+
+        let _ = try await [task, task2, task3]
+
+        // THEN
+        XCTAssertEqual(mockAPI.sendMethodCalledTimes, 1)
+    }
+    
+    func testConfigurationLoaderStartLoadingAfterFinishedLoading() async throws {
+        // GIVEN
+        mockAPI.entity = .configuration()
+        let loader = ConfigurationLoader(api: mockAPI)
+
+        // WHEN
+        _ = try await loader.fetch()
+        _ = try await loader.fetch()
+
+        // THEN
+        XCTAssertEqual(mockAPI.sendMethodCalledTimes, 2)
+    }
+    
+    func testConfigurationLoaderStartSecondLoadingIfFirstFailed() async throws {
+        // GIVEN
+        mockAPI.entity = .configuration()
+        mockAPI.errors = [NSError(domain: "", code: 0)]
+        let loader = ConfigurationLoader(api: mockAPI)
+
+        // WHEN
+        do {
+            _ = try await loader.fetch()
+            XCTFail("Error needs to be thrown.")
+        } catch {}
+        _ = try await loader.fetch()
+
+        // THEN
+        XCTAssertEqual(mockAPI.sendMethodCalledTimes, 2)
+    }
+}
+
+extension RemoteConfiguration {
```

Comment:
I think there is lots of legacy APIs and connections. I would suggest including the minimal URLs we need or even specifying a separate API for "next gen wallet" and sorting things out on the backend to avoid duplication.